### PR TITLE
Add challenge code

### PR DESCRIPTION
**Pull request by RenanTKN**

### CODE REVIEW

File: chall/str_replace.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#include <iostream>
+#include <string>
+using namespace std;
+
+string ReplaceString(string subject, const string& search, const string& replace) {
```
Comment (ataias) [inline chall/str_replace.cpp]: what about snake_case? The C++ book we are following recommends it I think.

File: chall/str_replace.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	size_t pos = 0;
+	while ((pos = subject.find(search, pos)) != std::string::npos) {
```
Comment (ataias) [inline chall/str_replace.cpp]: You area already using namespace std. Why not "string::npos" instead?

File: chall/str_replace.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	string solution = "I found the solution of this challenge!";
+	string str_to_replace = "solution";
+	cout << "Your output must be: '" << solution << "' to win:" << endl;
+	char str[100];
+	cin.getline(str, sizeof(str));
+	string output = ReplaceString(str, str_to_replace, "");
+	cout << "Your output is:" << endl;
+	cout << output << endl;
+
+	if (solution == output)
```
Comment (ataias) [inline chall/str_replace.cpp]: I think an interesting practice is always use curly braces, even if not needed. It is quite easy to extend code and then forget to add them later.

**Review by ataias — COMMENTED** [review]
I have some suggestions. Check them out.

File: chall/str_replace.cpp (hunk 1 above)
Comment (RenanTKN) [reply chall/str_replace.cpp]: I just copy and paste this function. Fixed.

File: chall/str_replace.cpp (hunk 2 above)
Comment (RenanTKN) [reply chall/str_replace.cpp]: Good catch. Fixed.

File: chall/str_replace.cpp (hunk 3 above)
Comment (RenanTKN) [reply chall/str_replace.cpp]: Very true, fixed.

**Pull request merged**